jetty-jaspi/src/main/java/org/eclipse/jetty/security/jaspi/modules/BaseAuthModule.java

```diff
@@ -22,15 +22,15 @@
 import javax.security.auth.callback.Callback;
 import javax.security.auth.callback.CallbackHandler;
 import javax.security.auth.callback.UnsupportedCallbackException;
-import javax.security.auth.message.AuthException;
-import javax.security.auth.message.AuthStatus;
-import javax.security.auth.message.MessageInfo;
-import javax.security.auth.message.MessagePolicy;
-import javax.security.auth.message.callback.CallerPrincipalCallback;
-import javax.security.auth.message.callback.GroupPrincipalCallback;
-import javax.security.auth.message.config.ServerAuthContext;
-import javax.security.auth.message.module.ServerAuthModule;
 
+import jakarta.security.auth.message.AuthException;
```

Comment:
I think the name of this class should be 'AbstractAuthModule` so we don't get confused with BASIC authentication. It should also be `abstract`.

Reply:
Most of the classes are simple implementations of the required Jakarta Authentication interfaces, so method doc should be inherited. 
The way Jakarta Authentication works is has a number of abstract-wrapper-factory-wrapper-layers which requires a bit of bridging code to jetty.  Since this does not support dynamically registering auth modules via a web admin gui, or any other profiles but the Servlet Container Profiles, most of these are only for briding jetty security to jakarta authentication.

It basically boils down to being able to call the AuthConfigFactory#registerConfigProvider (as jetty has no admin gui to manipulate ServerAuthModules)

In Jakarta Authentication 3.0 (Jakara EE 10) this will be further simplified by allowing one to register a ServerAuthModule directly on the AuthConfigFactory - avoiding some layers of wrapping (other than to support the wrapper-wrapper mode of registering auth modules).